erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -657,7 +657,7 @@ def get_gl_entries(self, warehouse_account):
 					item_account_wise_additional_cost.setdefault((d.item_code, d.name), {})
 					item_account_wise_additional_cost[(d.item_code, d.name)].setdefault(t.expense_account, 0.0)
 					item_account_wise_additional_cost[(d.item_code, d.name)][t.expense_account] += \
-						(t.amount * d.basic_amount) / total_basic_amount
+						(t.amount * d.basic_amount) / total_basic_amount if total_basic_amount else 0
```

Comment:
if `total_basic_amount = 0` then you should return outside the `for` loop, there is point in doing this unnecessary processing.

Reply:
Done